tests/vm_test.go

```diff
@@ -2109,3 +2068,19 @@ func stopVM(virtClient kubecli.KubevirtClient, vm *v1.VirtualMachine) *v1.Virtua
 
 	return vm
 }
+
+func patchVmRunStrategy(vm *v1.VirtualMachine, strategy v1.VirtualMachineRunStrategy) error {
+	vm, err := kubevirt.Client().VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	oldSpec := vm.Spec
+	newSpec := oldSpec.DeepCopy()
+	newSpec.Running = nil
+	newSpec.RunStrategy = &strategy
+
+	patchData, err := patch.GenerateTestReplacePatch("/spec", oldSpec, newSpec)
```

Comment:
This is pretty much `Update` and not the intended use case of a patch.

With a patch, we update the specific field we care about.
In this case, `Running` and `RunStrategy`.

And I do not see the need to use `GenerateTestReplacePatch`, because there seems to be no reason to perform a test. But if you think a test is needed as part of the patch, please explain what scenario it protects.

You need something like this:
```go
patches := patch.New(
	patch.WithAdd("/spec/running", nil),   // Maybe you need to pass here "`null`" instead of `nil`, not sure.
	patch.WithAdd("/spec/runStrategy", string(strategy)),
)
patchData, err := patches.GeneratePayload()
if err != nil {
	return err
}
_, err = kubevirt.Client().VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, patchData, metav1.PatchOptions{})
return err
```

Reply:
Thanks! understood. Done here: https://github.com/kubevirt/kubevirt/pull/11948.